pkg/azuredx/datasource.go

```diff
@@ -89,29 +87,27 @@ func NewDatasource(instanceSettings backend.DataSourceInstanceSettings) (instanc
 
 // QueryData is the primary method called by grafana-server
 func (adx *AzureDataExplorer) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+	ctx = adxusercontext.FromQueryReq(ctx, req)
```

Comment:
Can you make it explicit that you are adding the tokens here? I would rename this to:

```
	ctx = adxusercontext.WithRequestToken(ctx, req)
```

So it's clear what's doing (and easier to follow).

Reply:
any comment about this name change?

Reply:
I don't like current naming too but I had to come up with something to have three distinct funcs: for query request, call resource request and health check request. If Go would support overloads, it would be much easier...

What about this?
```
ctx := adxusercontext.WithUserFromQueryReq(ctx, ...)

ctx := adxusercontext.WithUserFromResourceReq(ctx, ...)

ctx := adxusercontext.WithUserFromHealthCheckReq(ctx, ...)
```

`adx user context with user from query request` etc.

Reply:
yeah, that sounds good too (as far as it's clear that it's adding user / auth info to the context)

Reply:
Renamed.